Solutions/Microsoft Defender for Cloud/Analytic Rules/CoreBackupDeletionwithSecurityAlert.yaml

```diff
@@ -20,38 +20,46 @@ relevantTechniques:
 query: |
   SecurityAlert
   | extend Extprop = parse_json(ExtendedProperties)
-  | extend Computer = iff(isnotempty(toupper(tostring(Extprop["Compromised Host"]))), toupper(tostring(Extprop["Compromised Host"])), tostring(parse_json(Entities)[0].HostName))
-  | extend Account = iff(isnotempty(tolower(tostring(Extprop["User Name"]))), tolower(tostring(Extprop["User Name"])), tolower(tostring(Extprop["user name"])))
-  | extend IpAddress = tostring(parse_json(ExtendedProperties).["IpAddress"]) 
-  | project TimeGenerated, AlertName, Computer, Account, IpAddress, ExtendedProperties
-  | extend timestamp = TimeGenerated, Account, MachineName = Computer, IpAddress
+  | mv-expand todynamic(Entities)
+  | extend HostName = iff(isnotempty(tostring(Extprop["Compromised Host"])), tolower(tostring(Extprop["Compromised Host"])), tolower(tostring(parse_json(Entities).HostName)))
```

Comment:
'Compromised Host' is sometimes showing more than one value e.g. 'SQLonMachine, SQL2017CRM'. Does it makes sense to do split on 'Compromised Host' with ',' and then mv-expand to list each value separately?

Reply:
Good catch. I've updated the query to handle this situation.